### ISSUE TITLE
eupspkg uses too many threads on massively multicore machines

### ISSUE THREAD

**Issue by mjuric** [RobertLuptonTheGood/eups#58]
@r-owen [reports](https://jira.lsstcorp.org/browse/DM-4734) that the [LSST afw](http://github.com/lsst/afw) package does not build reliably (if at all) on a linux box we have with 32 CPUs, each with 8 cores. The failure is that some unit tests fail with the following error:
```
OpenBLAS: pthread_creat error in blas_thread_init function. Error code:11
```
This is due to eupspkg attempting to launch too many threads, and hitting into OS limits.

The current workaround is to manually set `EUPSPKG_NJOBS` environmental variable on machines with many cores.

**Comment by mjuric** [comment]
@r-owen, could you see if the change on 95f841a68dc460db7d6f2501b4f9ad2ea125a84c fixes your problem (it's simple enough to be copied+pasted, just to test it).

**Comment by mjuric** [comment]
For posterity: there's been some discussion of this in the commit comments of 95f841a.

**Comment by mjuric** [comment]
@r-owen's done more debugging. We found that the original `pthread_create` problem is due to hitting the maximum number of processes (and POSIX threads count as processes Linux -- see `man pthread_create`, and the description of `EAGAIN` error). On RHEL (and CentOS) systems, this limit is 1024 for ordinary users, e.g.:
```
[mjuric@dev ~]$ ulimit -u
1024
```
@r-owen's shown the build works when this is lifted to a higher value (we've tried with 8192).

Depending on how a system's been configured, this limit may be changed by any process. I'll add a patch to `eupspkg.sh` to try to increase the limit to `4096` (which would mean ~128 processes per core, for a 32-core machine), and warn if it can't do that and detects a high core count (high `NJOBS`).
